contents/tutorials/astro-ab-tests.md

```diff
@@ -0,0 +1,380 @@
+
+Server-side rendering is a way to avoid this. This fetches the feature flag before the page loads on the client.
+
+To set this up, we must install and use [PostHog’s Node library](/libraries/node) (because we are making server-side requests).
+
+```bash
+npm install posthog-node
+```
+
+In the `src` folder, create a `posthog-node.js` file. This is where we set up the code to create the PostHog Node client. You can find both your API key and instance address in your [project settings](https://us.posthog.com/project/settings).
+
+```js file=src/posthog-node.js
+import { PostHog } from 'posthog-node';
+
+let posthogClient = null;
+
+export default function PostHogNode() {
+  if (!posthogClient) {
+    posthogClient = new PostHog('<ph_project_api_key>', {
+      host: '<ph_instance_address>',
+    });
+  }
+  return posthogClient;
+}
+```
+
+Next, we import `posthog-node.js` into `pages/index.astro`. Then we use it to fetch the feature flag and update the button text:
+
+```astro file=index.astro
+---
+import Layout from '../layouts/Layout.astro';
+import PostHogNode from '../posthog-node.js';
+
+let buttonText = 'No variant'
+try {
+  const distinctId = 'placeholder-user-id'
+  const enabledVariant = await PostHogNode().getFeatureFlag('my-cool-experiment', distinctId);
+  if (enabledVariant === 'control') {
+		buttonText = 'Control Variant';
+	} else if (enabledVariant === 'test') {
+		buttonText = 'Test Variant';
+	}
+} catch (error) {
+  buttonText = 'Error';
+}
+---
+<Layout>
+	<html lang="en">
+		<head>
+			<meta charset="utf-8" />
+			<link rel="icon" type="image/svg+xml" href="/favicon.svg" />
+			<meta name="viewport" content="width=device-width" />
+			<meta name="generator" content={Astro.generator} />
+			<title>Astro</title>
+		</head>
+		<body>
+			<h1>Astro A/B tests</h1>
+			<button class="main-button">{buttonText}</button>
+
+			<script>
+				const button = document.querySelector('.main-button');
+				button.addEventListener('click', () => {
+					window.posthog.capture('home_button_clicked')
+				});
+    	</script>	
+		</body>
+	</html>
+</Layout>
+```
+
+Now, when you refresh the page, the button text is already set when the page loads. 
+
+#### Setting the correct `distinctId`
+
+You may notice that we set `distinctId = 'placeholder-user-id'` in our flag call above. In production apps, to ensure you fetch the correct flag value for your user, `distinctId` should be set to their unique ID. 
+
+For logged-in users, you typically use their email as their `distinctId`. However, for logged-out users, you can use the `distinct_id` property from their PostHog cookie:
+
+```astro file=index.astro
+---
+import Layout from '../layouts/Layout.astro';
+import PostHogNode from '../posthog-node.js';
+
+const projectAPIKey = '<ph_project_api_key>';
+const cookie = Astro.cookies.get(`ph_${projectAPIKey}_posthog`);
+let buttonText = 'No variant'
+if (cookie && cookie.json().distinct_id) {
```

Comment:
I would love it if the cookie doesn't exist to use the client-side PostHog call to get the flag as a fallback (if possible 😅 )

Reply:
It's a good idea, but I think it'll make the tutorial code a bit convoluted